app/views/spree/admin/users/_api_fields.html.haml

```diff
@@ -1,17 +1,26 @@
 %fieldset.omega.six.columns
   %legend= t('spree.api.access')
+  = form_with(model: @user, url: spree.admin_user_path(@user)) do |form|
+    = form.check_box :show_api_key_view, onchange: "this.form.submit()"
+    = form.label :show_api_key_view, t('spree.api.toggle_api_key_view')
   - if @user.spree_api_key.present?
     .field
       = label_tag t('spree.api.key')
       = ":"
       = @user.spree_api_key
+
     .filter-actions.actions
-      = form_tag spree.clear_api_key_admin_user_path(@user), method: :put do
-        = button t('spree.api.clear_key'), 'icon-trash'
-      = form_tag spree.generate_api_key_admin_user_path(@user), method: :put do
-        = button t('spree.api.regenerate_key'), 'icon-refresh'
+      = form_with(model: @user, url: spree.admin_user_path(@user)) do |form|
+        = form.hidden_field :spree_api_key, value: nil
+        = form.button t('spree.api.clear_key'), class: 'icon-trash', icon: 'icon-trash'
+      = form_with(model: @user, url: spree.admin_user_path(@user)) do |form|
+        = form.hidden_field :spree_api_key, value: SecureRandom.hex(24)
```

Comment:
I was wondering if we should allow this. I guess that it's okay for admins.

Just an idea, don't implement that now: we could define a new attribute `:generate_api_key` on the model and define the method `generate_api_key=` which generates a new key when given a truthy value.

But I would probably prefer to use the new ApiKeysController here, too. Why not?